diff -dur libmba-0.9.1.ORIG/mktool.c libmba-0.9.1/mktool.c
--- libmba-0.9.1.ORIG/mktool.c	2005-04-28 02:11:24.000000000 -0600
+++ libmba-0.9.1/mktool.c	2007-04-28 12:14:07.000000000 -0600
@@ -462,11 +462,17 @@
 		}
 		if (arg_array[ARG_SOLNK1][0]) {
 			slibprintf(lname, arg_array[ARG_SOLNK1], libname, vers);
-			symlink(fname, lname);
+			if (symlink(fname, lname) < 0) {
+				ERR(errno, argv[argc - 1]);
+				return -1;
+			}
 		}
 		if (arg_array[ARG_SOLNK2][0]) {
 			slibprintf(lname, arg_array[ARG_SOLNK2], libname, vers);
-			symlink(fname, lname);
+			if (symlink(fname, lname) < 0) {
+				ERR(errno, argv[argc - 1]);
+				return -1;
+			}
 		}
 	} else {
 		if (doinstcmd(argv[argc - 1], argc - i - 1, argv + i, mode ? mode : 0644) == -1) {
